### PR TITLE
feat(Button): New and improved buttons

### PR DESCRIPTION
**Pull request by maraisr**
With this PR I aim to amend current implementation of buttons - with this, we get a better more optimized lean button component to help us future proof a bit better.

Example: 
image

### CODE REVIEW

File: lib/components/Anchor/stories.tsx [hunk 1]
```diff
@@ -18,11 +18,15 @@ storiesOf('Foundation|Typography/Anchor', module)
...
+		'withButton',
+		() => (
+			<Anchor
+				{...baseProps()}
+				is={Button}
+				to="./#eldorado"
+				icon={<Icon icon={PhoneIcon} size={16} />}
+			/>
+		),
+		{ chromatic: { disable: true } },
```
Comment (amir-zahedi) [inline lib/components/Anchor/stories.tsx]: Why did we need to disable chromatic in here?

File: lib/components/Button/Button.scss [hunk 2]
```diff
@@ -0,0 +1,241 @@
+@import '../../theme/tokens/facade';
+
+@mixin isState($state, $nest: false) {
```
Comment (amir-zahedi) [inline lib/components/Button/Button.scss]: We don't seem to be using the nest option anywhere

File: lib/components/Button/Button.scss [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	justify-content: center;
+	align-items: center;
+}
+
+// Sizes
+.small,
+.medium {
+	&.minimal {
+		min-width: auto;
+		//padding: 0;
```
Comment (amir-zahedi) [inline lib/components/Button/Button.scss]: Remove the commented out padding

File: lib/components/Button/Button.tsx [hunk 4]
```diff
@@ -1,135 +1,107 @@
...
+type AllowedChildren = string | IconType;
+
+interface Props
+	extends Pick<ButtonPrimitive, 'id' | 'onClick' | 'type' | 'className'>,
+		Pick<AriaAttributes, 'aria-label'> {
+	children: AllowedChildren | AllowedChildren[];
 	disabled?: boolean;
-	is?: ElementType;
-	isFullWidth?: boolean;
+	is?: ReactElement;
```
Comment (amir-zahedi) [inline lib/components/Button/Button.tsx]: Should we limit this to 'a' and 'button' only?

Comment (maraisr) [reply lib/components/Button/Button.tsx]: This is mainly here to support the `<Link />` react router will give us, etc...

File: lib/components/Button/Button.scss (hunk 2 above)
Comment (maraisr) [reply lib/components/Button/Button.scss]: Its more just there in case we need it in the future. Think I did use it at some point.

**Pull request merged**